### ISSUE TITLE
TypeError: Cannot read property 'length' of null

### ISSUE THREAD

**Issue by phet-steele** [phetsims/unit-rates#200]
```
Uncaught TypeError: Cannot read property 'length' of null
    at ShoppingItemNode.getParent (Node.js?bust=1499874450676:777)
    at ShoppingItem.animationCompletedCallback (ShoppingItemDragHandler.js?bust=1499874450676:192)
    at ShoppingItem.step (URMovable.js?bust=1499874450676:122)
    at FruitScene.step (ShoppingScene.js?bust=1499874450676:299)
    at ShoppingCategory.step (ShoppingCategory.js?bust=1499874450676:57)
    at ShoppingModel.step (ShoppingModel.js?bust=1499874450676:92)
    at Sim.stepSimulation (Sim.js?bust=1499874450676:830)
    at Sim.runAnimationLoop (Sim.js?bust=1499874450676:770)
```

**Comment by pixelzoom** [comment]
I suspect that in ShoppingNode, `visibleObserver` has been queued to be called before `dispose` is called.  And then when  `visibleObserver` is called, it attempts to perform operations on `self` post-disposal.  So the assertion in Node `getParent` (line 777) fails:

```js
776   getParent: function() {
777      assert && assert( this._parents.length <= 1, 'Cannot call getParent on a node with multiple parents' );
778      return this._parents.length ? this._parents[ 0 ] : null;
779    },
```

@jonathanolson Can you confirm that `_parents` is scuttled when Node is disposed?

**Comment by pixelzoom** [comment]
Btw... I have been unable to reproduce this with `?ea&screens=1&fuzzMouse`.  And I'm unclear on why it would only happen infrequently.

**Comment by jessegreenberg** [comment]
Is this issue related to https://github.com/phetsims/scenery/issues/647 (SimpleDragHandler disposal)?

**Comment by pixelzoom** [comment]
Hmmm... could be.

**Comment by jonathanolson** [comment]
> Can you confirm that _parents is scuttled when Node is disposed?

That's correct (checks for things like `removeChild( disposedNode )`).

> Is this issue related to phetsims/scenery#647 (SimpleDragHandler disposal)?

I don't see an immediate connection. This assertion failure is stemming from a step() instead of a pointer event.

**Comment by pixelzoom** [comment]
Fixed in the above commit, using the same approach as in https://github.com/phetsims/graphing-lines/issues/83 (testing `this.disposed`).  Tested using `?ea&fuzzMouse`.

Since this is the second sim where I've had to do this, I'm a bit concerned that there is a general pattern to this problem. I will create a separate issue to discuss.

@phet-steele please verify.

**Comment by pixelzoom** [comment]
@phet-steele is still seeing this after pulling, took ~1 hour to hit in fuzz test.

https://github.com/phetsims/scenery/issues/649 is going to make this workaround unnecessary, so let's hold off on this until that's done.

**Comment by pixelzoom** [comment]
@phet-steele This has been fixed in phetsims/scenery#649, and the workaround in unit-rates has been reverted. Please pull scenery and unit-rates to verify.

**Comment by phet-steele** [comment]
Fuzzed master for an hour and saw no errors, closing.
